examples/nidaqmx/analog-input.py

```diff
@@ -110,15 +112,15 @@ def check_for_warning(response):
     stop_task_response = client.StopTask(nidaqmx_types.StopTaskRequest(task=task))
     check_for_warning(stop_task_response)
     print(
-        f"Acquired {len(response.read_array)} samples",
-        f"({response.samps_per_chan_read} samples per channel)",
+        f"Acquired {len(read_response.read_array)} samples",
+        f"({read_response.samps_per_chan_read} samples per channel)",
     )
 except grpc.RpcError as rpc_error:
     error_message = str(rpc_error.details() or "")
-    for key, value in rpc_error.trailing_metadata() or []:  # type: ignore
-        if key == "ni-error":
-            details = value if isinstance(value, str) else value.decode("utf-8")
-            error_message += f"\nError status: {details}"
+    for entry in rpc_error.trailing_metadata() or []:
```

Comment:
Did you test that this works at run time? I thought we tried this and had problems with inconsistency between the sync and aio APIs.

Reply:
grpcio's sync `MetadataType` is `Sequence[Tuple[str, Union[str, bytes]]]` which does not allow using named `key` and `value` fields: https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_typing.py#L39

I think the underlying cython code uses a plain old tuple at run time: https://github.com/grpc/grpc/blob/master/src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi#L61

grpc-stubs uses a class with `key` and `value` fields, which doesn't support destructuring. This seems like a bug in grpc-stubs. A `Tuple` or `NamedTuple` would support destructuring. https://github.com/shabbyrobe/grpc-stubs/blob/master/grpc-stubs/__init__.pyi#L416

Reply:
> This is in main,

Hmm, ok, looks like I made that change in https://github.com/ni/grpc-device/pull/930. I'm pretty sure I tested it at run time, not just mypy.

Reply:
Yeah, sorry. Should've clarified up front in the PR description what this PR was about. Added that now.